File: models/Message.js
const mongoose = require("mongoose");

const messageSchema = new mongoose.Schema(
  {
    chatId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Chat",
      required: true,
    },
    senderId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
    messageType: {
      type: String,
      enum: ["text", "image", "file","voice","video"],
      default: "text",
    },
    content: {
      type: String,
      required: true,
    },
    duration: {
      type: Number, 
    },
    seenBy: [{ type: mongoose.Schema.Types.ObjectId, ref: "User" }]
  },
  {
    timestamps: true,
    versionKey: false,
  }
);

module.exports = mongoose.model("Message", messageSchema);

File: lib/model/Message.dart
class Message {
  final String? id;
  final String? senderId;
  final String? chatId;
  final String? messageType;
  final String? content;
  final DateTime? timestamp;
  final List<String>? seenBy;
  final int? duration;

  Message({this.id, this.messageType, this.chatId, this.content,this.timestamp,this.duration,this.seenBy,this.senderId});

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['_id'],
      messageType: json['messageType'],
      chatId: json['chatId'],
      senderId: json['senderId'],
      content: json['content'],
      duration: json['duration'],
      timestamp: DateTime.parse(json['createdAt']),
      seenBy: List<String>.from(json['seenBy']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'messageType': messageType,
      'chatId': chatId,
      'content': content,
      'timestamp': timestamp,
      'seenBy': seenBy,
      'duration': duration,
      'senderId': senderId,
    };
  }
}
